awx/api/serializers.py

```diff
@@ -4230,6 +4235,67 @@ def get_summary_fields(self, obj):
         d['recent_notifications'] = self._recent_notifications(obj)
         return d
 
+    def validate_messages(self, messages):
+        if messages is None:
+            return None
+
+        error_list = []
+        collected_messages = []
+
+        if not isinstance(messages, dict):
+            error_list.append(_("Expected dict for 'messages' field, found {}".format(type(messages))))
+        else:
+            for event in messages:
+                if event not in ['started', 'success', 'error']:
+                    error_list.append(_("Event '{}' invalid, must be one of 'started', 'success', or 'error'").format(event))
+                    continue
+                event_messages = messages[event]
+                if event_messages is None:
+                    continue
+                if not isinstance(event_messages, dict):
+                    error_list.append(_("Expected dict for event '{}', found {}").format(event, type(event_messages)))
+                    continue
+                for message_type in event_messages:
```

Comment:
FYI:

```>>> for k, v in {'a': 1, 'b': 2, 'c': 3}.items():
...   print("key [{}] value [{}]".format(k, v))
...
key [a] value [1]
key [b] value [2]
key [c] value [3]
```

vs. `message = event_messages[message_type]` so something like:
`for message_type, message in event_messages.items():`